api/v1alpha6/conversion.go

```diff
@@ -580,3 +591,37 @@ func Convert_v1alpha7_BindingProfile_To_Map_string_To_Interface(in *infrav1.Bind
 	}
 	return nil
 }
+
+func Convert_v1alpha7_OpenStackClusterStatus_To_v1alpha6_OpenStackClusterStatus(in *infrav1.OpenStackClusterStatus, out *OpenStackClusterStatus, s conversion.Scope) error {
+	err := autoConvert_v1alpha7_OpenStackClusterStatus_To_v1alpha6_OpenStackClusterStatus(in, out, s)
+	if err != nil {
+		return err
+	}
+
+	// Router and APIServerLoadBalancer have been moved out of Network in v1alpha7
+	if in.Router != nil || in.APIServerLoadBalancer != nil {
+		if out.Network == nil {
+			out.Network = &Network{}
+		}
+
+		out.Network.Router = (*Router)(in.Router)
+		out.Network.APIServerLoadBalancer = (*LoadBalancer)(in.APIServerLoadBalancer)
```

Comment:
`Router` or `APIServerLoadBalancer` still could be nil, right?
If so, we could have to separate `if` blocks. One for `in.Router != nil`, one for `in.APIServerLoadBalancer != nil`.

Reply:
I recall the fuzzer tests beat me up a bit on this one. The issue is that in either case you need to check if Network is nil, so I combined them. They're no-ops if the fields are nil anyway.